test/unit/search_controller.test.js

```diff
@@ -0,0 +1,510 @@
+import { expect } from 'chai';
```

Comment:
Not sure if this is still WIP but let's test new things with TS too.

Reply:
@arnautov-anton Does it make sense to have ts declarations if all would be mocks?

Reply:
I mean I saw that the controller takes some generic parameter so it'd be good to test that too. Also tests if functions which rely on this generic accept specific parameter types correctly and what not. Having fully-typed tests just adds extra level of safety even if fully or partially mocked - meaning changing some types after the fact should be caught too.

Reply:
I still am not sure, whether TS version would make sense. All the search source classes are based on a single abstract class that does not care about the generic type - there is no impact on logic. Could you give me an example please?